includes/Feed/FeedProductFormatter.php

```diff
@@ -0,0 +1,236 @@
+			$parent_attribute_values = array(
+				'gallery_urls'       => $gallery_urls,
+				'default_variant_id' => $variation_id,
+				'item_group_id'      => \WC_Facebookcommerce_Utils::get_fb_retailer_id( $parent_product->woo_product ),
+			);
+
+			foreach ( $variants_for_group as $variant ) {
+				if ( isset( $variant['product_field'], $variant['options'] ) ) {
+					$parent_attribute_values[ $variant['product_field'] ] = $variant['options'];
+				}
+			}
+
+			// Cache product group variants.
+			$attribute_variants[ $parent_id ] = $parent_attribute_values;
+
+		} else {
+
+			$parent_attribute_values = $attribute_variants[ $parent_id ];
+		}
+
+		$variants_for_item   = $fb_product->prepare_variants_for_item( $product_data );
+		$variant_feed_column = array();
+
+		foreach ( $variants_for_item as $variant_array ) {
+
+			static::format_variant_for_feed(
+				$variant_array['product_field'],
+				$variant_array['options'][0],
+				$parent_attribute_values,
+				$variant_feed_column
+			);
+		}
+
+		if ( isset( $product_data['custom_data'] ) && is_array( $product_data['custom_data'] ) ) {
+
+			foreach ( $product_data['custom_data'] as $product_field => $value ) {
+
+				static::format_variant_for_feed(
+					$product_field,
+					$value,
+					$parent_attribute_values,
+					$variant_feed_column
+				);
+			}
+		}
+
+		if ( ! empty( $variant_feed_column ) ) {
+			$product_data['variant'] = '"' . implode( ',', $variant_feed_column ) . '"';
+		}
+
+		if ( isset( $parent_attribute_values['gallery_urls'] ) ) {
+			$product_data['additional_image_urls'] = array_merge( $product_data['additional_image_urls'], $parent_attribute_values['gallery_urls'] );
+		}
+
+		if ( isset( $parent_attribute_values['item_group_id'] ) ) {
+			$product_data['item_group_id'] = $parent_attribute_values['item_group_id'];
+		}
+
+		$product_data['default_product'] = $parent_attribute_values['default_variant_id'] == $fb_product->id ? 'default' : '';
+
+		return $product_data;
+	}
+
+
+	/**
+	 * Format additional urls into the feed requirements.
+	 *
+	 * @since 2.6.0
+	 * @param array $product_image_urls Array of urls to images.
+	 * @return string Stringified list of urls.
+	 */
+	private function format_additional_image_url( $product_image_urls ) {
+		// Returns the top 5 additional image urls separated by a comma according to feed api rules.
+		$product_image_urls = array_slice(
+			$product_image_urls,
+			0,
+			self::MAX_IMAGES_PER_PRODUCT
+		);
+		if ( $product_image_urls ) {
+			return '"' . implode( ',', $product_image_urls ) . '"';
+		} else {
+			return '';
+		}
+	}
+
+	/**
+	 * Change WC price format to Feed price format.
+	 *
+	 * @since 2.6.0
+	 * @param float  $value Value to format.
+	 * @param string $currency Currency of value.
+	 * @return string Formatted price value.
+	 */
+	private function format_price_for_feed( $value, $currency ) {
+		return (string) ( round( $value / (float) 100, 2 ) ) . $currency;
```

Comment:
Didn't notice anything wrong on FB but based on [API reference](https://developers.facebook.com/docs/marketing-api/catalog-batch/reference#product-item) the format should _with a space between cost and currency._. Fixed in 62d2cdca74470df02e0591d458cb55b1e3efb734